client/src/utils/Store.js

```diff
@@ -103,6 +109,91 @@ export default class Store {
     }
   }
 
+  /**
+   * Syncs the models on the client with the models on the server. If the
+   * client has models the server doesn't have, those are added to the server.
+   */
+  syncModels = async () => {
+    try {
+      // Get the model names from the server. Model names are used as equality
```

Comment:
So, for a particular user, the model names must all be unique. However, different users could have the same model names -- will this break things?

Reply:
It will not because the only model names returned from the server are for the particular user session from which the request is coming from. So when hitting that endpoint you only get model names for models you have locally on that browser.